### PR TITLE
MES-4028- Category C schema done

### PR DESCRIPTION
**Pull request by amobarak**
# Description and relevant Jira numbers

All CAT C schemas are done, updated common and changed B/B+E as well.
Branch containing all C sub categories is: feature/catCSubCategories

## Pull Request checklist

- [ ] [WIP] tag removed from PR title
- [x] PR has an understandable description

## Git feature branch checklist

- [ ] branch name comply with our branching strategy
- [ ] git branch contains relevant JIRA ticket number
- [ ] branch rebased against the latest develop

## Sign off process checklist

- [x] Code has been tested manually
- [ ] Tested by QA
- [ ] PO's approval
- [ ] Reviewers selected in Github
- [ ] PR link added to JIRA

### CODE REVIEW

File: mes-test-schema/categories/B/partial.d.ts [hunk 1]
```diff
@@ -132,3 +160,12 @@ export interface QuestionResult {
...
   outcome?: "P" | "DF" | "S" | "D";
 }
+/**
+ * This interface was referenced by `PartialTestResultCatBSchema`'s JSON-Schema
+ * via the `definition` "vehicleDetails".
+ */
+export interface VehicleDetails {
+  schoolCar?: SchoolCar;
+  dualControls?: DualControls;
+  [k: string]: any;
```
Comment (rossfauldsbjss) [inline mes-test-schema/categories/B/partial.d.ts]: I think the [k: string]: any; shouldn't be here.  I think it needs AdditionalProperties to be set to false.
(this was a fix Ammar applied to something else, so have a chat if it's not clear)

File: mes-test-schema/categories/C/partial.json [hunk 2]
```diff
@@ -2,88 +2,124 @@
...
-          "type": "object"
+        "businessAddress": {
+          "$ref": "#/definitions/address"
+        },
+        "businessName": {
+          "$ref": "#/definitions/businessName"
+        },
+        "businessTelephone": {
+          "$ref": "#/definitions/businessTelephone"
         }
```
Comment (rossfauldsbjss) [inline mes-test-schema/categories/C/partial.json]: business name and telephone exist at multiple levels... not sure it should (see lines 80-92)

File: mes-test-schema/categories/C/partial.json [hunk 3]
```diff
@@ -93,96 +129,38 @@
...
-            "type": "object"
-          },
-          "type": "array"
-        },
-        "showMeTellMeComments": {
-          "$ref": "../Common/index.json#/definitions/faultComments"
-        },
-        "tellMeQuestions": {
-          "items": {
-            "additionalProperties": false,
```
Comment (rossfauldsbjss) [inline mes-test-schema/categories/C/partial.json]: Why have show me questions and tell me questions been removed (actually it was called show me question and should be show me questions

Comment (rossfauldsbjss) [reply mes-test-schema/categories/C/partial.json]: removing comment. it is in vehicle checks.

File: mes-test-schema/categories/C/partial.json (hunk 2 above)
Comment (amobarak) [reply mes-test-schema/categories/C/partial.json]: This is done so businessName and businessTelephone have the correct type in the partial.d.ts file, the only way to keep the type of it is to have it as a local definition and then use that definition in the nesting.

File: mes-test-schema/categories/B/partial.d.ts (hunk 1 above)
Comment (amobarak) [reply mes-test-schema/categories/B/partial.d.ts]: Will update this one

File: mes-test-schema/categories/C/partial.json (hunk 2 above)
Comment (amobarak) [reply mes-test-schema/categories/C/partial.json]: When I do what you recommended, the following is what happens:
`export interface Candidate {
  businessAddress?: Address;
  /**
   * Name of the business the candidate relates to
   */
  businessName?: string;
  /**
   * Telephone number of the business the candidate relates to
   */
  businessTelephone?: string;
}`
You end up losing the type of businessName and businessTelephone.

**Pull request merged**